src/webnn/native/dml/ContextDML.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 The WebNN-native Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef WEBNN_NATIVE_DML_CONTEXT_DML_H_
+#define WEBNN_NATIVE_DML_CONTEXT_DML_H_
+
+#include "webnn/native/Context.h"
+
+#include "common/Log.h"
+#include "dml_platform.h"
+#include "webnn/native/Graph.h"
+
+namespace webnn::native::dml {
+
+    struct CommandRecorderDML {
+        ComPtr<IDMLDevice> device;
+        ComPtr<ID3D12Device> D3D12Device;
+        ComPtr<IDMLCommandRecorder> commandRecorder;
+        ComPtr<ID3D12CommandQueue> commandQueue;
+        ComPtr<ID3D12CommandAllocator> commandAllocator;
+        ComPtr<ID3D12GraphicsCommandList> commandList;
+    };
+
+    class Context : public ContextBase {
+      public:
+        static ContextBase* Create(ComPtr<IDXGIAdapter1> adapter, bool useDebugLayer);
+        ~Context() override = default;
+
+        CommandRecorderDML GetCommandRecorderDML() const {
+            return mCommandRecorderDML;
+        };
+
+      private:
+        Context(ComPtr<IDXGIAdapter1> adapter, bool useDebugLayer);
+        HRESULT Initialize();
+
+        GraphBase* CreateGraphImpl() override;
+
+        CommandRecorderDML mCommandRecorderDML;
```

Comment:
unit: rename `CommandRecorder` to be consistent.

Reply:
CommandRecorder mCommandRecorder?

Reply:
Yup.